pkg/provision/providers/vm/network.go

```diff
@@ -0,0 +1,41 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package vm
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+// getVmnetInterfaceName returns the name of the interface that will be assigned to the next vmnet interface.
+// The name is assigned incrementing by one, starting from "bridge100".
+func getVmnetInterfaceName(allCurrentInterfaces []string) (string, error) {
+	vmnetInterfaceFound := false
+	largestVmnetIfIndex := 100
+	vmnetInterfaceRegex := regexp.MustCompile(`\bbridge(1\d\d)\b`)
```

Comment:
let's move this into a `var` declaration so it doesn't get compiled every time this function is called

Reply:
we could also now write a test for this

Reply:
nvm, saw the test below

Reply:
the good practice for regex:

```go
// global var
var vmnetInterfaceRegex = sync.OnceValue(func() *regex.Regex { return regexp.MustCompile(...) } )
```

this ensures it's compiled once and not compiled until first fused

Reply:
Ah genius! Thank you both!